org.eclipse.winery.frontends/app/tosca-management/src/app/wineryLicenseModule/wineryLicense.service.ts

```diff
@@ -51,34 +67,166 @@ export class WineryLicenseService {
         );
     }
 
-    loadLicenses(): Observable<string[]> {
+    getLicenseTextLocal(name: string): string {
+        const theLicense = this._allLicences.find(item => item.LICENCE_NAME.toLowerCase() === name.toLowerCase());
+        if (theLicense !== null && theLicense !== undefined) {
+            return theLicense.licenceText;
+        } else {
+            return 'License not found!';
+        }
+    }
+
+    loadLocalLicenses(): Observable<string[]> {
         // create an array of observables to execute them later on parallel.
-        const allObservables = this._allLicences.map(license => {
+        const allObservables = this._allLicences.map((license) => {
             const headers = new HttpHeaders({ 'Accept': 'text/plain' });
             return this.http.get(license.LICENSE_FILE_URL, { headers: headers, responseType: 'text' })
                 .pipe(
-                    map((result: string) => {
+                    map((result) => {
                         license.licenceText = result;
                         return result;
                     })
                 );
         });
-
         // execute the observables on parallel
         return forkJoin(allObservables);
     }
 
-    getLicenseText(name: string): string {
-        const theLicense = this._allLicences.find(item => item.LICENCE_NAME.toLowerCase() === name.toLowerCase());
+    getLicenseNames(): string[] {
+        return this._allLicences.map(license => license.LICENCE_NAME);
+    }
 
-        if (theLicense !== null && theLicense !== undefined) {
-            return theLicense.licenceText;
-        } else {
-            return 'License not found!';
+    /*
+    License engine calls
```

Comment:
Maybe we should split them into two different services?

Reply:
I agree. Done.